rspec-core/lib/rspec/core/configuration.rb

```diff
@@ -1773,6 +1773,40 @@ def warnings?
         $VERBOSE
       end
 
+      if defined?(::Warning) && ::Warning.respond_to?(:[]=)
+        # Set Ruby deprecation warnings on or off.
+        def deprecation_warnings=(value)
+          @deprecation_warnings_set = true
+          ::Warning[:deprecated] = value
+        end
+
+        # @return [Boolean] Whether or not ruby warnings are enabled.
+        def deprecation_warnings?
+          ::Warning[:deprecated]
+        end
+
+        # @private
+        def deprecation_warnings_set?
+          @deprecation_warnings_set || ::Warning[:deprecated]
```

Comment:
This is causing a warning because if it hasn't been set to true its not initialised.

Reply:
Ah right,, that's indeed an issue until Ruby 2.7. Fied it, thanks.